stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImpl.kt

```diff
@@ -152,6 +155,33 @@ internal class QueryChannelsControllerImpl(
                 refreshChannel(event.cid)
             }
         }
+
+        if (event is UserPresenceChangedEvent) {
+            refreshMembersStateForUser(event.user)
+        }
+    }
+
+    private suspend fun refreshMembersStateForUser(newUser: User) {
+        val userId = newUser.id
+
+        val affectedChannels = _channels.value
+            .values
+            .filter { channel -> channel.users().map { user -> user.id }.contains(userId) }
```

Comment:
You don't need to map all users, you can use the following filter
```suggestion
            .filter { channel -> channel.users().any { it.id == userId} }
```